### DISCUSSION TITLE
The mac virtual environment venv terminal cannot directly use the shortcut command pip

### DISCUSSION THREAD

**Discussion by Alan3344** [pypa/virtualenv#2631] [Q&A] [answered]
create venv `python -m venv .venv`
Python 3.11.5
pip 23.2.1 from /opt/homebrew/lib/python3.11/site-packages/pip (python 3.11)
zsh 5.9 (arm-apple-darwin22.1.0)
Kernel: arm64 Darwin 22.5.0
Using pip directly will not point to the pip of the virtual environment
![image](https://github.com/pypa/virtualenv/assets/60773912/b0b7c658-382d-48d9-9e5d-1bbbd7789ef2)
```
(.venv) ➜  recatpy_ whereis pip
pip: /Users/alan/me/recatpy_/.venv/bin/pip
```
I also tried running these two commands, is there any difference
- `/Users/alan/me/recatpy_/.venv/bin/pip list` Ok
- `.venv/bin/pip list` OK
- `pip list` NO (it shows all packages)

**Comment by Alan3344** [comment]
I tried using `export=/Users/alan/me/recatpy_/.venv/bin/pip` using `pip list` doesn't work
Then I thought of using the `alias` command, and then it took effect. Is this script for activating the virtual environment a little problem?
Then I tested...
![image](https://github.com/pypa/virtualenv/assets/60773912/5bf1a829-1e35-44c5-a350-714e7319bf5e)

**Comment by Alan3344** [comment]
I add this function in my `~/.zshrc`

```bash
pyvenv() {
    envfolder=$1
    if [ -z "$envfolder" ]; then
        envfolder=.venv
    fi
    envdir=`pwd`/$envfolder &&
    python -m venv $envdir &&
    if [ -d "$envdir" ]; then
        echo alias pip=$envdir/bin/pip >> $envdir/bin/activate &&
        source $envdir/bin/activate &&
        echo -e "\e[33m"$envdir"\e[0m" has been created as virtual environment!
        return
    fi
}
```
Use `pyvenv` or `pyvenv <venv dirname>` to create `venv`

**Reply by gaborbernat** [reply, marked as answer]
venv is not virtualenv, https://virtualenv.pypa.io/en/latest/user_guide.html#user-guide

**Comment by gaborbernat** [comment]
This discussion has an accepted answer. As noted, `venv` (Python's built-in module) is not the same as `virtualenv`. For `venv`-specific questions, please refer to Python's documentation: https://docs.python.org/3/library/venv.html

Closing as answered.
